File: Haskell/2022/02.hs
import Data.Char
import Lib

type Input = [(Int, Int)]

main :: IO ()
main = aoc 2022 2 setup solve1 solve2 ["1"]

solve1 :: Input -> Int
solve1 = sum . map (\(x, y) -> (y - x + 1) `mod` 3 * 3 + y)

solve2 :: Input -> Int
solve2 = sum . map (\(x, y) -> (y + x) `mod` 3 + y * 3 - 2)

setup :: String -> Input
setup = map (\[a, _, b] -> (ord a - 64, ord b - 87)) . lines

File: Python/2022/02.py
from lib import *

input = read_input(2022, 2)

part1 = 0
part2 = 0
for line in input.splitlines():
    a, b = line.split()
    a = ord(a) - ord("A")
    b = ord(b) - ord("X")

    w = (a - b) % 3
    if w == 0:
        part1 += 3 + b + 1
    elif w == 1:
        part1 += b + 1
    elif w == 2:
        part1 += 6 + b + 1

    w = (a + b - 1) % 3
    if b == 0:
        part2 += w + 1
    elif b == 1:
        part2 += 3 + w + 1
    elif b == 2:
        part2 += 6 + w + 1

print(part1)
print(part2)
